src/index.js

```diff
@@ -212,39 +212,37 @@ export default async function microbundle(inputOptions) {
 
 	if (options.watch) {
 		const onBuild = options.onBuild;
-		return new Promise((resolve, reject) => {
-			stdout(
-				blue(
-					`Watching source, compiling to ${relative(
-						cwd,
-						dirname(options.output),
-					)}:`,
+
+		stdout(
+			blue(
+				`Watching source, compiling to ${relative(
+					cwd,
+					dirname(options.output),
+				)}:`,
+			),
+		);
+
+		return steps.map(options => {
```

Comment:
I wonder if people know the order of steps and which watcher is which, maybe it doesn't matter.

Reply:
What do you think about reducing the steps to an `entry file name` -> `watcher` object instead of an array? 

Its hard to know which use cases for this might pop up in the future :), i personally only need the array, but a map might be useful for sure - I am open for alternatives if you have an idea :D

Reply:
I made some changes: 
- I reverted my change which removed the Promise from microbundle watch. 
- Microbundles promise with enabled watch will now return an object like:

```js
{
  watchers: {
    "/some/absolute/input/filename.js": RollupWatcher
  }
}
```